### PR TITLE
Ccg api

### PR DESCRIPTION
**Pull request by dayesouza**

### CODE REVIEW

File: toolkit/compare_case_groups/api.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        )
+
+        self.filtered_df = filter_df(self.model_df, filters)
+
+        grouped_df = build_grouped_df(self.filtered_df, groups)
+
+        attributes_df = build_attribute_df(self.filtered_df, groups, aggregates)
+
+        temporal_df = pl.DataFrame()
+        if temporal is not None and temporal != "":
```
Comment (andresmor-ms) [inline toolkit/compare_case_groups/api.py:100]: this can be replaced by `if temporal`

File: toolkit/compare_case_groups/api.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    def _format_list(self, items, bold=True, escape_colon=False) -> str:
+        formatted_items = []
+        for item in items:
+            if escape_colon:
+                item = item.replace(":", "\\:")
+            if bold:
+                item = f"**{item}**"
+            formatted_items.append(item)
+        return "[" + ", ".join(formatted_items) + "]"
```
Comment (andresmor-ms) [inline toolkit/compare_case_groups/api.py:126]: is the idea to return a json list here? you could use `json.dumps(formatted_items)

File: toolkit/compare_case_groups/build_dataframes.py [hunk 3]
```diff
@@ -133,14 +118,13 @@ def build_attribute_df(
     )
 
 
-def filter_df(main_df, filters: list[str]):
-    filtered_df = main_df.copy()
-
+def filter_df(main_df: pl.DataFrame, filters: list[str]) -> pl.DataFrame:
     if len(filters) == 0:
```
Comment (andresmor-ms) [inline toolkit/compare_case_groups/build_dataframes.py:122]: can be replaced by `if filters:`.

In python empty sequences (len == 0) is False

File: toolkit/compare_case_groups/api.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+            ]
+        else:
+            columns = [
+                *lower_groups,
+                "group_count",
+                "group_rank",
+                "attribute_value",
+                "attribute_count",
+                "attribute_rank",
+            ]
```
Comment (andresmor-ms) [inline toolkit/compare_case_groups/api.py:66]: what about:

```python
columns = [g.lower() for g in self.groups]

columns.extend(["group_count", "group_rank", "attribute_value", "attrubite_count", "attribute_rank"]  # You could move this into a constant :)

if self.temporal:
    columns.extend([f"{self.temporal}_window", ...])


```

that way you avoid having the else statement and avoid some duplicated lines

File: toolkit/compare_case_groups/build_dataframes.py (hunk 3 above)
Comment (dayesouza) [reply toolkit/compare_case_groups/build_dataframes.py:122]: Actually this if filter can be removed since it will not go into the loop anyway right

Comment (andresmor-ms) [reply toolkit/compare_case_groups/build_dataframes.py:122]: you are right :)

File: toolkit/compare_case_groups/api.py (hunk 4 above)
Comment (dayesouza) [reply toolkit/compare_case_groups/api.py:70]: Yes, Thanks!

File: toolkit/compare_case_groups/api.py (hunk 2 above)
Comment (dayesouza) [reply toolkit/compare_case_groups/api.py:130]: it's to return a string actually

**Pull request merged**